Assets/Scripts/ObjectPool/ObjectPool.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectPool : MonoBehaviour
+{
+    [SerializeField] private int _spawnAmount;
+    [SerializeField] private GameObject _object;
+
+    private GameObject _parent;
+
+    private Queue<GameObject> _objectPool = new Queue<GameObject>();
+
+    private void Awake()
+    {
+        _parent = new GameObject(_object.name);
+        GenerateNewAmount();
+    }  
+
+    /// <summary>
+    /// Call this function to generate a new amount of object in a pool
+    /// </summary>
+    private void GenerateNewAmount()
+    {
+        for (int i = 0; i < _spawnAmount; i++)
+            ReturnObject(Instantiate(_object));        
+    }
+
+    /// <summary>
+    /// Call this function to get an object out of the pool
+    /// </summary>
+    /// <returns>The first object of pool</returns>
+    public GameObject GetObject()
```

Comment:
Maybe nicer to remove the "Object" part of the `GetObject()` method name. You're working with an `**Object**pool`, so it is already obvious you're returning and getting objects

Reply:
Same for the `Return()` method